model/Concept.php

```diff
@@ -640,6 +641,9 @@ public function getProperties()
         }
 
         $ret = $this->removeDuplicatePropertyValues($ret, $duplicates);
+
+        $ret = $this->vocab->getConfig()->getShowNotationAsProperty() !== null ? $this->removeNotationFromProperties($ret, $this->vocab->getConfig()->getShowNotationAsProperty()) : $ret ;
```

Comment:
Rather than first adding and later removing the notation properties, I think it would be better to avoid adding them to `$ret` in the first place (if the configuration setting is false). That should be done within the foreach block the starts on line 578. So I suggest doing that there (something like: if getShowNotationAsProperty() returns false, and the property type is skos:notation, skip that property) and removing this line.

Reply:
I think we should reconsider the construction of long conditional statements

Prefer option 1 to option 2 or something else?

https://github.com/NatLibFi/Skosmos/blob/fba94659c41bec4a466c182c8713333f2cb4a00d/model/Concept.php#L578-L595

Guidelines:
https://www.php-fig.org/psr/psr-12/
_Expressions in parentheses MAY be split across multiple lines, where each subsequent line is indented at least once. When doing so, the first condition MUST be on the next line. The closing parenthesis and opening brace MUST be placed together on their own line with one space between them. Boolean operators between conditions MUST always be at the beginning or at the end of the line, not a mix of both._

```
<?php

if (
    $expr1
    && $expr2
) {
    // if body
} elseif (
    $expr3
    && $expr4
) {
    // elseif body
}

```

Or should we keep it separate from changes in functionality as @osma you mentioned here: https://github.com/NatLibFi/Skosmos/pull/1198#discussion_r704414499

Reply:
I'm sorry, I now see that I gave you bad advice yesterday. Even better than adjusting the foreach loop on line 578 (as you've now done) would be to do the check even earlier, around line 511. Currently there's a check to see if the property is in $this->DELETED_PROPERTIES and if it is, the entire property is skipped.

Perhaps the most elegant option would be doing something like this:

```php
$deleted = $this->DELETED_PROPERTIES;
if (!$this->vocab->getConfig()->getShowNotationAsProperty()) {
  // delete notations as well unless configured to show them
  $deleted[] = 'skos:notation';
}

// adjust the check on line 511 to use $deleted instead of $this->DELETED_PROPERTIES
if (!in_array($prop, $this->DELETED_PROPERTIES) || ($this->isGroup() === false && $prop === 'skos:member')) {
  ...
```

Reply:
I suppose it is ready now:

https://github.com/NatLibFi/Skosmos/blob/f0a932db45c114309565e5ba1d25aaaf7a3acc59/model/Concept.php#L512-L517